GoogleForCreators/web-stories-wp
# Editor: New design for color picker

## Context

This updates the color picker to the new slimmer, leaner design.

This also adds a ton of new configuration options for the color input and the color picker, that ar required for the color inputs in the new design menu.

And while working on these elements, I made a number of other optimizations as well (including optimizing CSS and DOM structure).

**Note:** Screenshots are incoming (once I can run a virtual machine again when I get a proper charger for my laptop), so please just clone and check for yourself. The storybook stories are the best way to play around with this (Stories Editor > Components > Form > Color > *)

## Summary

* A lot of the design decisions (spacing, etc) has been made on the fly, because the full color picker was not designed by UX. Figma files only existed for the basic color picker, not the custom color picker.

## Relevant technical choices

* Updated swatches to only render 1 CSS class in total, not 1 per swatch (by moving the actual swatch pattern styles to `style` rather than the class)

## To-do

* [x] Add (comprehensive!) storybook stories for color input in both design panel (left and right) as well as new design menu.
* [x] Add before and after screenshots

## User-facing changes

The color picker as opened from the design menu is a lot slimmer in general now.

| Picker | Saved colors? | Before | After |
|-|-|-|-|
| Basic | Yes | ![Screen Shot 2022-02-24 at 16 34 27](https://user-images.githubusercontent.com/637548/155611779-f8284a31-593d-49a4-8174-0d7f36f5db01.png) | ![Screen Shot 2022-02-24 at 16 32 17](https://user-images.githubusercontent.com/637548/155611767-80d84eca-743d-4024-8926-907e467a30a0.png) |
| Basic | No | ![Screen Shot 2022-02-24 at 16 34 35](https://user-images.githubusercontent.com/637548/155612297-51b1fc0e-2490-4fa4-86ae-f067be7382a8.png) | ![Screen Shot 2022-02-24 at 16 33 06](https://user-images.githubusercontent.com/637548/155612161-06e90cbb-2eb6-4f4d-8c9d-9e80c95ade40.png) |
| Custom, solid | Yes | ![Screen Shot 2022-02-24 at 16 34 31](https://user-images.githubusercontent.com/637548/155612057-d4c44fcf-d48c-4351-8e76-7845f7241f29.png) | ![Screen Shot 2022-02-24 at 16 33 21](https://user-images.githubusercontent.com/637548/155612107-625973b9-5f0b-42d3-bfe3-dd8447dda344.png) |
| Custom, solid | No | ![Screen Shot 2022-02-24 at 16 34 40](https://user-images.githubusercontent.com/637548/155612036-aeee01d2-3eb7-487b-bc18-cf0558109996.png) | ![Screen Shot 2022-02-24 at 16 33 15](https://user-images.githubusercontent.com/637548/155612135-53419ed1-e454-48fa-a0eb-29ba2635b335.png) |
| Custom, gradient | Yes | ![Screen Shot 2022-02-24 at 16 34 20](https://user-images.githubusercontent.com/637548/155612075-4c436188-c6b4-4376-af75-a841c06240f3.png) | ![Screen Shot 2022-02-24 at 16 32 05](https://user-images.githubusercontent.com/637548/155612179-83dff986-927d-4778-a8aa-07fa75a906bc.png) |

## Non-user-facing changes

Color picker as opened from the new design menu looks like the following (note this is still behind feature flag).

| Picker | Eyedropper | Gradient | Screenshot |
|-|-|-|-|
| Basic | No | No | ![Screen Shot 2022-02-24 at 16 43 48](https://user-images.githubusercontent.com/637548/155613048-98233bba-2135-4a7d-8c33-d806948460a9.png) |
| Custom | No | No | ![Screen Shot 2022-02-24 at 16 43 56](https://user-images.githubusercontent.com/637548/155613204-bc3ec524-dfbf-4391-882d-e1cffbd1d293.png) |
| Basic | Yes | No | ![Screen Shot 2022-02-24 at 16 44 37](https://user-images.githubusercontent.com/637548/155613216-8b4f5c05-710f-473c-b02a-7ad2e87506df.png) |
| Custom | Yes | No | ![Screen Shot 2022-02-24 at 16 44 41](https://user-images.githubusercontent.com/637548/155613227-e846ed79-e61a-4795-8e06-6529eaa630da.png) |
| Custom | Yes | Yes | ![Screen Shot 2022-02-24 at 16 45 04](https://user-images.githubusercontent.com/637548/155613238-6443aba7-eb19-4a35-a536-76231aae13f2.png) |

## Testing Instructions



This PR can be tested by following these steps:

1. Open the color picker for page backgrounds, borders, text colors, text backgrounds, and shapes.
2. Observe, that everything works as before.

## Checklist




---

Fixes #10626
